### PR TITLE
Fixes #12996 - Tests run without errors after removing a plugin

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

File: test/functional/hosts_controller_test.rb [hunk 1]
```diff
@@ -962,7 +962,7 @@ class Host::Valid < Host::Managed; end
   test 'template_used returns templates with interfaces' do
     @host.setBuild
     nic=FactoryGirl.create(:nic_managed, :host => @host)
-    attrs = @host.attributes.except('id', 'created_at', 'updated_at')
+    attrs = @host.attributes.except('id', 'created_at', 'updated_at').slice(*Host::Managed.accessible_attributes.to_a)
```
Comment (ohadlevy) [inline test/functional/hosts_controller_test.rb]: dry?

**Comment by orrabin** [comment]
@ohadlevy I fixed your comment.

File: test/functional/hosts_controller_test.rb [hunk 2]
```diff
@@ -1029,7 +1033,7 @@ def test_submit_multiple_rebuild_config_pessimistic
 
     group2 = FactoryGirl.create(:hostgroup, :compute_profile => compute_profiles(:two))
 
-    attrs = host.attributes.except('id', 'created_at', 'updated_at')
+    attrs = host.attributes.except('id', 'created_at', 'updated_at').slice(*Host::Managed.accessible_attributes.to_a)
```
Comment (ohadlevy) [inline test/functional/hosts_controller_test.rb]: maybe pass the host object instead?

Comment (orrabin) [inline test/functional/hosts_controller_test.rb]: I missed this one, changing it to be like puppetclass test.

**Comment by dLobatog** [comment]
Merged as d30a7ff3175d754fa21525a9f9a6d0c2951e5c53, thanks @orrabin!

**Pull request closed without merging**